arduino-ide-extension/src/browser/theia/workspace/workspace-service.ts

```diff
@@ -82,13 +91,25 @@ export class WorkspaceService extends TheiaWorkspaceService {
     }
   }
 
+  override open(uri: URI, options?: WorkspaceOptions): void {
+    this.optionsToAppendToURI = options;
```

Comment:
Please find a way to not store any possible commands as a state in the workspace service.

Reply:
What do you suggest? I couldn't think of another way of doing it without reimplementing `WorkspaceService.open` (which I would avoid)

Reply:
> reimplementing WorkspaceService.open (which I would avoid)

I doubt we can do it without a custom implementation. For a long-term solution, we should propose a change in Theia to pass any optional data to `WorkspaceService#openNewWindow`.